test/codeu/chat/util/TokenizerTest.java

```diff
@@ -0,0 +1,35 @@
+package codeu.chat.util;
+
+import java.io.IOException;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+/**
+ * Created by Jiahui Chen on 5/24/2017.
+ */
+public class TokenizerTest {
+    @Test
+    public void testWithNoQuotes() throws IOException{
+        final Tokenizer noQuotesSimple = new Tokenizer("1 two 3 4");
+        assertEquals("1", noQuotesSimple.next());
+        assertEquals("two", noQuotesSimple.next());
+        assertEquals("3", noQuotesSimple.next());
+        assertEquals("4", noQuotesSimple.next());
+        assertEquals(null, noQuotesSimple.next());
+
+        final Tokenizer noQuotesEmpty = new Tokenizer("");
+        assertEquals(null, noQuotesEmpty.next());
+    }
+
+    @Test
+    public void testWithQuotes() throws IOException{
+        final Tokenizer withQuotesSimple = new Tokenizer("\"hello world\" \"how are you\"");
```

Comment:
I thought of edge cases like that too, but the technical activities doc didn't mention any edge cases and also the example test cases didn't cover edge cases so I figured it was okay to leave it as is.